src/csharp/Grpc.Core/Channel.cs

```diff
@@ -223,6 +223,17 @@ protected override async Task ShutdownAsyncCore()
             shutdownTokenSource.Cancel();
 
             var activeCallCount = activeCallCounter.Count;
+
+            // Part of fix for https://github.com/grpc/grpc/issues/28153 memory leak
+            if (activeCallCount > 0)
+            {
+                // There may be a race condition when disposing of a call and then
+                // immediately closing the channel. This reduces the chance although
+                // not completely eliminating it. It gives callbacks a chance to happen.
+                await Task.Delay(200);
```

Comment:
this seems very arbitrary. We can't have a random delay in the shutdown procedure like this.
Is this `. only about avoiding the warning message to be printed (while the call is disposed properly) or 2. is that actually needed for the call to be disposed at all?
If it's 1., I think we can live with the message being printed if the channel is abandoned too early. If it's 2. that's a bigger problems since we can't have the call sometimes ending up disposed properly and sometimes not.